roles/rsyslog/templates/output_remote_files.j2

```diff
@@ -13,16 +13,25 @@ template(
   string="{{ __remote_log_path }}"
 )
 ruleset(name="{{ __rsyslog_output.name }}"
+{%   if __rsyslog_output.queue.type is not defined %}
         queue.type="{{ logging_server_queue_type }}"
+{%   endif %}
+{%   if __rsyslog_output.queue.size is not defined %}
         queue.size="{{ logging_server_queue_size }}"
-        queue.workerThreads="{{ logging_server_threads }}") {
+{%   endif %}
+{%   if __rsyslog_output.queue.workerthreads is not defined %}
+        queue.workerThreads="{{ logging_server_threads }}"
+{%   endif %}
```

Comment:
Not really happy with this code tho, it should not test for the parameter in lower case, the test should be case insensitive.

Reply:
% if __rsyslog_output.queue is defined %}
{%   set __queue_options = {} %}
{%   for option in __rsyslog_output.queue.keys() %}
{%     set _none = __queue_options.update({ option | lower : true }) %}
{%   endfor %}
{% endif %}
.....
{%   if __queue_options.type is not defined %}
        queue.type="{{ logging_server_queue_type }}"
{%   endif %}
{%   if  __queue_options.type.size is not defined %}
        queue.size="{{ logging_server_queue_size }}"
{%   endif %}
{%   if __queue_options.workerthreads is not defined %}
        queue.workerThreads="{{ logging_server_threads }}"
{%   endif %}

This would make the test case insensitive.

Reply:
nice - lgtm

Reply:
Well this is a special case just for the output_remote_files.j2 template since that template was already using some custom queue options to get better performance than with default settings.